swarm_copy/tools/kg_morpho_features_tool.py

```diff
@@ -0,0 +1,357 @@
+    "Section Bif Lengths": ["min", "max", "median", "mean", "std"],
+    "Section Branch Orders": ["min", "max", "median", "mean", "std"],
+    "Section Bif Branch Orders": ["min", "max", "median", "mean", "std"],
+    "Section Term Branch Orders": ["min", "max", "median", "mean", "std"],
+    "Section Path Distances": ["min", "max", "median", "mean", "std"],
+    "Section Taper Rates": ["min", "max", "median", "mean", "std"],
+    "Local Bifurcation Angles": ["min", "max", "median", "mean", "std"],
+    "Remote Bifurcation Angles": ["min", "max", "median", "mean", "std"],
+    "Partition Asymmetry": ["min", "max", "median", "mean", "std"],
+    "Partition Asymmetry Length": ["min", "max", "median", "mean", "std"],
+    "Sibling Ratios": ["min", "max", "median", "mean", "std"],
+    "Diameter Power Relations": ["min", "max", "median", "mean", "std"],
+    "Section Radial Distances": ["min", "max", "median", "mean", "std"],
+    "Section Term Radial Distances": ["min", "max", "median", "mean", "std"],
+    "Section Bif Radial Distances": ["min", "max", "median", "mean", "std"],
+    "Terminal Path Lengths": ["min", "max", "median", "mean", "std"],
+    "Section Volumes": ["min", "max", "median", "mean", "std"],
+    "Section Areas": ["min", "max", "median", "mean", "std"],
+    "Section Tortuosity": ["min", "max", "median", "mean", "std"],
+    "Section Strahler Orders": ["min", "max", "median", "mean", "std"],
+    "Soma Surface Area": ["raw"],
+    "Soma Radius": ["raw"],
+    "Number Of Sections Per Neurite": ["min", "max", "median", "mean", "std"],
+    "Total Length Per Neurite": ["min", "max", "median", "mean", "std"],
+    "Total Area Per Neurite": ["min", "max", "median", "mean", "std"],
+    "Total Height": ["raw"],
+    "Total Width": ["raw"],
+    "Total Depth": ["raw"],
+    "Number Of Neurites": ["raw"],
+    "Soma Number Of Points": ["N"],
+}
+
+
+class FeatRangeInput(BaseModel):
+    """Features Range input class."""
+
+    min_value: float | int | None = None
+    max_value: float | int | None = None
+
+
+class FeatureInput(BaseModel):
+    """Class defining the scheme of inputs the agent should use for the features."""
+
+    label: LABEL
+    compartment: (
+        Literal["Axon", "BasalDendrite", "ApicalDendrite", "NeuronMorphology", "Soma"]
+        | None
+    ) = Field(
+        default=None,
+        description=(
+            "Compartment of the cell. Leave as None if not explicitely stated by the"
+            " user"
+        ),
+    )
+    feat_range: FeatRangeInput | None = None
+
+    @model_validator(mode="before")
+    @classmethod
+    def check_if_list(cls, data: Any) -> dict[str, str | list[float | int] | None]:
+        """Validate that the values passed to the constructor are a dictionary."""
+        if isinstance(data, list) and len(data) == 1:
+            data_dict = data[0]
+        else:
+            data_dict = data
+        return data_dict
+
+
+class MorphoFeatureInput(BaseModel):
```

Comment:
This name is very similar to `MorphologyFeatureInput` in `morphology_features_tool.py`

Reply:
I added the KG prefix as with the other classes in this module